webapp/Sigbuilder-graph.js

```diff
@@ -0,0 +1,1872 @@
+                if(y_check[1].length > 7){
+                    y = y.toFixed(7);
+                }
+            }
+            if(i == (graphParameters.graphPoints.length-1)){
+                x_arr += x + "]";
+                y_arr += y + "]";
+            }else{
+                x_arr += x + ";";
+                y_arr += y + ";";
+            }
+        }
+        }else{
+            x_arr = "[0]";
+            y_arr = "[0]";
+        }
+        var sort_x_array = graphParameters.original_X_Points.slice(0);
+        sort_x_array.sort();
+        var compare_x_array = "["+sort_x_array.toString().replace(/,/g, ';')+"]";
+        if(compare_x_array.trim() == x_arr.trim()){
+            x_arr = "["+graphParameters.original_X_Points.toString().replace(/,/g, ';')+"]";
+            y_arr = "["+graphParameters.original_Y_Points.toString().replace(/,/g, ';')+"]";
+        }
+        var propertiesObject = {
+            id: cell.id
+                };
+        propertiesObject["xx"] = x_arr;
+        propertiesObject["yy"] = y_arr;
+        propertiesObject["Method"] = (graphParameters.mtd).toString();
+        propertiesObject["PeriodicOption"] = (graphParameters.PeriodicOption).toString();
+        propertiesObject["graf"] = "y";
+        wind.destroy();
+        get_parameters_wind_sigbuilder.show();
+        check_call = 2;
+        cell.blockInstance.instance.set(propertiesObject);
+    };
+    check_call = 1;
+}
+
+function autoscaleFunctionalityForGraph(drag_sig_chart,graphParameters){
+    //Added for postive/maximum value autoscale functionality
+        var max_x_value_new = (drag_sig_chart.xAxis[0].getExtremes().dataMax).toFixed(1);
+        var min_x_value_new = (drag_sig_chart.xAxis[0].getExtremes().dataMin).toFixed(1);
+        var max_y_value_new = (drag_sig_chart.yAxis[0].getExtremes().dataMax).toFixed(1);
+        var min_y_value_new = (drag_sig_chart.yAxis[0].getExtremes().dataMin).toFixed(1);
+        var max_x = drag_sig_chart.xAxis[0].getExtremes().max;
+        var max_y = drag_sig_chart.yAxis[0].getExtremes().max;
+        var min_y = drag_sig_chart.yAxis[0].getExtremes().min;
+        var diff_x = (((Math.abs(max_x_value_new - min_x_value_new))/100)*10).toFixed(2);
+        var diff_y = (((Math.abs(max_y_value_new - min_y_value_new))/100)*10).toFixed(2);
+        if(Math.abs(max_x - max_x_value_new) < parseFloat(diff_x)){
+            max_x = parseFloat(parseFloat(max_x) + parseFloat(diff_x)).toFixed(1);
+            drag_sig_chart.xAxis[0].update({
+                max: parseFloat(max_x)
+            });
+        }
+        if(Math.abs(max_y - max_y_value_new) < parseFloat(diff_y)){
+            max_y = parseFloat(parseFloat(max_y) + parseFloat(diff_y)).toFixed(1);
+            drag_sig_chart.yAxis[0].update({
+                max: parseFloat(max_y)
+            });
+        }
+        //Added for negative/minimum value autoscale functionality for y axis
+        if(Math.abs(min_y - min_y_value_new) < parseFloat(diff_y)){
```

Comment:
```suggestion
        if (diff_y > 0.1) {
```
is if correct? just check of diff_y > 0.1